services/text_extractor_service.py

```diff
@@ -14,17 +15,31 @@ class TextExtractorService:
     def __init__(self, db, executor):
         self.executor = executor
         self.stored_recipes_service = StoredRecipesService(db)
+        # You can change maxmemory=100mb and maxmemory-policy=allkeys-lfu from redis-cli.
+        self.redis = redis.Redis(host='redis-10060.c135.eu-central-1-1.ec2.cloud.redislabs.com', port=10060,
+                                 username='default', password='CWqOy4UlreQiOgo8lenpSc6ZMnCvTA8x', decode_responses=True)
```

Comment:
we need a small design change here.
The redis client should be injected rather than given the responsibility of the text extractor service to build it . 

try to think of a way to separate this logic. where should the logic of "how do I insert/get the data and from/to where"

some hints:

Think how would you design the code in a way that changing the database/cache would't force us change the code of text_extractor_service?

Reply:
I think with some modifications we can create a caching service with a single instance call from app that is used by both text_extractor and redis_blacklist.
Unless it's better practice to create a separated instance for each usage?

It will probably reduce a lot of code duplications.